interface/src/app/signature/signature.js

```diff
@@ -92,8 +92,71 @@ angular.module('adage.signature', [
   }]
 )
 
-.directive('highRangeExp', ['$http', '$log', '$state', 'ActivityDigits',
-  function($http, $log, $state, ActivityDigits) {
+// A service that defines the specifications of embedded vega-lite widget.
+// It is used by "highRangeExp" directive.
+.service('EmbedSpecService', [function() {
+  // sample2index: a dictionary that maps a sample's ID to the index
+  // of this sample's activity value in "EmbedSpecService.spec.data.values".
+  // It will make the toggling of "highlight" flag faster.
+  sample2index = {};
+
+  this.mode = 'vega-lite';
+  this.actions = false;
```

Comment:
`mode` and `actions` are properties only required by the vega-lite specification.